dattri/task.py

```diff
@@ -168,26 +177,36 @@ def get_grad_target_func(
                 If the input is a tensor, the corresponding element should be the
                 dimension of the tensor.
             layer_name (Optional[Union[str, List[str]]]): This has not been supported.
+            index (Optional[int]): The index of the checkpoint to be loaded, only
+                needed when layer_name is not None.
 
         Returns:
             Callable: The function that computes the gradient of the target function.
-
-        Raises:
-            NotImplementedError: If layer_name is not None.
         """
+        # first add decorator that handles the layer_name
+        target_func = self.target_func
         if layer_name is not None:
```

Comment:
Here the condition is "not None" while in the function `get_param`, the default value of `layer_name` is 0.  We need to make it consistent.

Reply:
Also, do we want to support a list of index?